### PR TITLE
feat: OTEL logging support

### PR DESCRIPTION
**Pull request by mhennoch**

### CODE REVIEW

**Comment by mhennoch** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mhennoch** [comment]
recheck

**Comment by mhennoch** [comment]
recheck

**Comment by mhennoch** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/logging/index.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  let resource = detectResource();
+
+  const serviceName =
+    options.serviceName ||
+    getNonEmptyEnvVar('OTEL_SERVICE_NAME') ||
+    resource.attributes[SemanticResourceAttributes.SERVICE_NAME];
+
+  if (!serviceName) {
+    diag.warn(
+      'service.name attribute is not set, your service is unnamed and will be difficult to identify. ' +
```
Comment (seemk) [inline src/logging/index.ts:93]: Might want to specify that it is from the logging part, else we'll get 2 identical warnings (1 from tracing)

**Review by seemk — APPROVED** [review]

**Pull request merged**